cmd/helm/installer/install.go

```diff
@@ -113,40 +114,65 @@ func createDeployment(client extensionsclient.DeploymentsGetter, opts *Options)
 
 }
 
-// deployment gets the deployment object that installs Tiller.
-func deployment(opts *Options) (*v1beta1.Deployment, error) {
-	return generateDeployment(opts)
+// Deployment gets the deployment object that installs Tiller.
+func Deployment(opts *Options) (*v1beta1.Deployment, error) {
+	dep, err := generateDeployment(opts)
+	if err != nil {
+		return nil, err
+	}
+	dep.TypeMeta = metav1.TypeMeta{
```

Comment:
shouldn't this be part of `generateDeployment`? :)

Reply:
Yeah I think it probably could be. If you look at the original code, the `TypeMeta` information is only included on the object when generating manifests for `-o` and `--debug` and is injected manually as strings: https://github.com/kubernetes/helm/blob/master/cmd/helm/init.go#L230. So, in order to minimize the potential impact, this PR follows that same pattern: TypeMeta is only included when generating the manifests as strings and not when actually sending the object to the Kubernetes api. 

That said, I don't see why including TypeMeta on the objects sent to the Kubernetes api would cause a problem. What do you think?

Reply:
Ooh good point. I didn't fully see that `generateDeployment` was sorta/kinda double dipping: once for the actual object being shipped to kubernetes on `helm init`, and once for the string output. This is fine. :+1: